pkg/machine/ignition/ignition.go

```diff
@@ -239,6 +241,23 @@ func (ign *DynamicIgnition) GenerateIgnitionConfig() error {
 		ignSystemd.Units = append(ignSystemd.Units, qemuUnit)
 	}
 
+	// Only AppleHv with Apple Silicon can use Rosetta
+	if ign.VMType == define.AppleHvVirt && runtime.GOARCH == "arm64" {
+		rosettaUnit := Systemd{
+			Units: []Unit{
+				{
+					Enabled: BoolToPtr(true),
```

Comment:
Thanks for linking the unit files, my assumption is that toggle relies on the fact the vfkit makes the rossetta virtiofs mount tag available or not, is this correct?

I have rather strong feeling about the these unit files. I really dislike the `|| true` part as it basically hides all errors from the unit/systemd POV so it makes debugging errors hard than it should be.
Second your units are hooked into the multi-user target which could be to late if containers are auto started at boot before your unit finished.

I will do some test and then post some (to me at least) better versions

Reply:
Can you test https://github.com/containers/podman-machine-os/pull/8? With this we would not even enable the unit at ignition time as it can be enabled by default.

Reply:
I built and tested machine-os locally.
Unfortunately, the following code does not work.
```
# This service can always be enabled as it has conditions to only run on apple arm hardware
RUN systemctl enable rosetta-activation.service
```
When I start the Podman machine, rosetta-activation.service is disabled.
```
root@localhost:/var/home/core# systemctl status rosetta-activation.service 
○ rosetta-activation.service - Activates Rosetta if necessary
     Loaded: loaded (/etc/systemd/system/rosetta-activation.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: inactive (dead)
root@localhost:/var/home/core# ls -d /etc/systemd/system/systemd-binfmt.service.wants/rosetta-activation.service
ls: cannot access '/etc/systemd/system/systemd-binfmt.service.wants/rosetta-activation.service': No such file or directory
```
I have previously encountered the same problem so I have enabled the service in Ignition.

If the service is enabled, rosetta-activation.sh works fine after rework.

Reply:
Thanks for testing, I guess instead of calling systemctl enable I could just add the symlink manually in the Containerfile, then we do not have to enable it here with ignition.

Reply:
I modified the Containerfile, but the result is the same.
Also, with and without the `ostree container commit` command, the result is the same.
```
# This service can always be enabled as it has conditions to only run on apple arm hardware
# RUN systemctl enable rosetta-activation.service
RUN mkdir -p /etc/systemd/system/systemd-binfmt.service.wants/ && \
    ln -s /etc/systemd/system/rosetta-activation.service /etc/systemd/system/systemd-binfmt.service.wants/rosetta-activation.service && \
    ostree container commit
```

`/etc/systemd/system/systemd-binfmt.service.wants` directory seems to disappear after Podman machine Ignition.
```
root@localhost:/var/home/core# systemctl status rosetta-activation.service 
○ rosetta-activation.service - Activates Rosetta if necessary
     Loaded: loaded (/etc/systemd/system/rosetta-activation.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/service.d
             └─10-timeout-abort.conf
     Active: inactive (dead)
root@localhost:/var/home/core# ls -d /etc/systemd/system/systemd-binfmt.service.wants
ls: cannot access '/etc/systemd/system/systemd-binfmt.service.wants': No such file or directory
root@localhost:/var/home/core# ls -d /etc/systemd/system/systemd-
systemd-homed.service.wants/    systemd-journald.service.wants/ systemd-repart.service     
```

Reply:
Weird, thanks for checking this. I guess I remove it then from the machine-os PR then we stick to enabling it here